### DISCUSSION TITLE
Unable to get segmentation past 0% on GeoTIFFs

### DISCUSSION THREAD

**Discussion by jsolly** [opengeos/segment-geospatial#164] [Q&A]
## Context
I am attempting to segment a very small sample of harmonized Sentinel-Landsat imagery. I have included this GeoTIFF inside my container:
https://github.com/easierdata/Ukraine_E2E_App/blob/main/ukraine_sample.tif

Here is my dockerfile on[ DockerHub](https://hub.docker.com/repository/docker/jsolly/ukraine_e2e_app_amd64/general) and the [whole repo with all the files and code is here](https://github.com/easierdata/Ukraine_E2E_App/tree/main).

```dockerfile
FROM giswqs/segment-geospatial

USER root
WORKDIR /

COPY fetch_model_weights.py .
COPY step4_segment_granules.py .
COPY ukraine_sample.tif .

# Change permission of the script to make it readable and executable
RUN chmod 755 /step4_segment_granules.py
RUN chmod 755 /fetch_model_weights.py

# Download model weights during build time so they are cached for later use
RUN python3 /fetch_model_weights.py

# Run the segment_granules.py script when the container launches
CMD ["python3", "step4_segment_granules.py"]
```

Here is the Python Script that is running
```py
from samgeo import SamGeo


def segment_granules(mosaic_file_path: str) -> None:
    sam = SamGeo(
        model_type="vit_h",
        checkpoint="sam_vit_h_4b8939.pth",
        sam_kwargs=None,
    )

    sam.generate(source=mosaic_file_path, output="HLS_segment.tif", batch=True)
    sam.tiff_to_gpkg(
        tiff_path="HLS_segment.tif", output="HLS_segmentation_polygons.gpkg"
    )


if __name__ == "__main__":
    segment_granules(mosaic_file_path="ukraine_sample.tif")
```

When I run it with:
```
docker run --rm jsolly/ukraine_e2e_app_amd64:july28
```
The output stays stuck at 0% 0/64. I let it run for 15 mins and it stayed at 0% the whole time while maxing out my machine's resources. I am on an M1 MacBook Pro with a GPU, so this very small amount of imagery, (84,240 pixels)

## Thoughts
I've tried this docker container with many different kinds of imagery, but they all stay stuck at 0%. Since this used to work, I am at a loss to what is wrong. If someone can try out my Docker Image with their imagery, it would help me narrow down the issue!

To create the imagery to pass into the model, I followed [this process](https://github.com/easierdata/Ukraine_E2E_App/blob/main/step3_mosaic_granule_bands.sh) to mosaic the imagery and make sure the values were between 0-255.

**Comment by giswqs** [comment]
It works fine locally on my end. I have not tried docker. 

https://samgeo.gishub.org/examples/automatic_mask_generator_hq/

![image](https://github.com/opengeos/segment-geospatial/assets/5016453/9e592685-35a3-42aa-917d-7f89ac7e24b7)

![image](https://github.com/opengeos/segment-geospatial/assets/5016453/a68a8acf-5b4c-4d5c-b4b1-75a09856c6c9)

**Comment by jsolly** [comment]
Thanks for checking ukraine_sample.tif!
If you could try the Docker image, that would be great!

I need to fetch the model weights during build time because the environment I am in does not allow calls to external resources when the container is running.

**Comment by jsolly** [comment]
Hi @giswqs ! Did you get a chance to check out the Docker image?

I am mainly pinging you because I am using your image from DockerHub and the code is straight from the [satellite image segmentation section](https://samgeo.gishub.org/examples/satellite/#segment-anything-model-for-geospatial-data) of the segment-geospatial doc page.

I tried running the [docker image with the model weights part taken out](https://hub.docker.com/layers/jsolly/ukraine_e2e_app_amd64/nofetchweights/images/sha256-6594ed5bdcbfa6d74a90369dd12b9feaea74097452301a07f5722b0ce45819a6?context=repo) and it's constantly failing about 20-30% through downloading the weights with this error
> Connection broken: IncompleteRead


See -> `jsolly/ukraine_e2e_app_amd64:nofetchweights`

**Reply by giswqs** [reply]
I am not a Docker user. I can't help much here. 

The Dockerfile simply installs the packages.  I don't see any particular settings that would cause the issue. 
https://github.com/opengeos/segment-geospatial/blob/main/Dockerfile

You can also try out this Dockerfile. 
https://github.com/Prindle19/segment-everything/blob/master/Dockerfile
